api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/hiv/EligibleForFastrackPatientsCalculation.java

```diff
@@ -44,55 +51,72 @@
     static ConceptService conceptService = Context.getConceptService();
+    static PatientService patientService = Context.getPatientService();
+
 
     @Override
     public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> parameterValues, PatientCalculationContext context) {
 
-        Concept TCAdate = Dictionary.getConcept(Dictionary.RETURN_VISIT_DATE);
+        String hivConsultationService = "885b4ad3-fd4c-4a16-8ed3-08813e6b01fa";
+        AppointmentsService appointmentsService = Context.getService(AppointmentsService.class);
         Program hivProgram = MetadataUtils.existing(Program.class, HivMetadata._Program.HIV);
         Set<Integer> alive = Filters.alive(cohort, context);
         Set<Integer> inHivProgram = Filters.inProgram(hivProgram, alive, context);
         Set<Integer> ltfu = CalculationUtils.patientsThatPass(calculate(new LostToFollowUpCalculation(), cohort, context));
         CalculationResultMap ret = new CalculationResultMap();
 
-        for(Integer ptId: cohort){
 
+        for(Integer ptId: cohort){
             Integer tcaPlus30days = 0;
-            Integer tcaConcept = 5096;
-            Date tcaDate = null;
             Date tcaObsDate = null;
+            Date appointmentDate = null;
             boolean patientInHivProgram = false;
-
             boolean eligible = false;
-            Date currentDate =new Date();
+            Date currentDate = new Date();
+            AppointmentSearchRequest appointmentSearchRequest = new AppointmentSearchRequest();
+            appointmentSearchRequest.setPatientUuid(patientService.getPatient(ptId).getUuid());
+            appointmentSearchRequest.setStatus(AppointmentStatus.Scheduled);
+            appointmentSearchRequest.setStartDate(currentDate);
+            List<Appointment> appointments = appointmentsService.search(appointmentSearchRequest);
+      
+            // Check if patient has a future appointment and get the appointment date
+            Optional<Appointment> futureAppointment = appointments.stream()
+                .filter(appointment -> appointment.getService().getUuid().equals(hivConsultationService))
+                .findFirst();
 
-         //With Greencard TCA more than 30 days
-            Encounter lastFollowUpEncounter = EmrUtils.lastEncounter(Context.getPatientService().getPatient(ptId), Context.getEncounterService().getEncounterTypeByUuid("a0034eee-1940-4e35-847f-97537a35d05e"));   //last greencard followup form
+            if (futureAppointment.isPresent()) {
+                appointmentDate = futureAppointment.get().getStartDateTime();
+                
+            }
+
+            //With Greencard TCA more than 30 days
+            Encounter lastFollowUpEncounter = EmrUtils.lastEncounter(patientService.getPatient(ptId), Context.getEncounterService().getEncounterTypeByUuid("a0034eee-1940-4e35-847f-97537a35d05e"));   //last greencard followup form
```

Comment:
We don't need this anymore. We already have a migration to bring all these TCA to the appointments model